slides/intro-extra-recipes.qmd

```diff
@@ -0,0 +1,378 @@
+\ \ step_zv(all_predictors()) %>%  
+\ \ step_log(distance) %>%  
+\ \ step_normalize(all_numeric_predictors())
+:::
+
+<br>
+
+Start by calling `recipe()` to denote the data source and variables used
+
+## What is a recipe?
+
+:::{style="font-family: 'Source Code Pro', monospace; font-size: 0.8em;"}
+recipe(tip ~ ., data = taxi_train) %>%  
+\ \ [step_unknown]{style="color: #CA225E;"}(all_nominal_predictors()) %>%  
+\ \ [step_dummy]{style="color: #CA225E;"}(all_nominal_predictors()) %>%  
+\ \ [step_zv]{style="color: #CA225E;"}(all_predictors()) %>%  
+\ \ [step_log]{style="color: #CA225E;"}(distance) %>%  
+\ \ [step_normalize]{style="color: #CA225E;"}(all_numeric_predictors())
+:::
+
+<br>
+
+Specify what actions to take by adding `step_*()`s.
+
+## What is a recipe?
+
+:::{style="font-family: 'Source Code Pro', monospace; font-size: 0.8em;"}
+recipe(tip ~ ., data = taxi_train) %>%  
+\ \ step_unknown([all_nominal_predictors()]{style="color: #CA225E;"}) %>%  
+\ \ step_dummy([all_nominal_predictors()]{style="color: #CA225E;"}) %>%  
+\ \ step_zv([all_predictors()]{style="color: #CA225E;"}) %>%  
+\ \ step_log([distance]{style="color: #CA225E;"}) %>% 
+\ \ step_normalize([all_numeric_predictors()]{style="color: #CA225E;"})
+:::
+<br>
+
+Use {tidyselect} and recipes-specific selectors to denote affected variables.
+
+## Using a recipe
+
+:::{style="font-family: 'Source Code Pro', monospace; font-size: 0.8em;"}
+rec_spec <- recipe(tip ~ ., data = taxi_train) %>%  
+\ \ step_unknown(all_nominal_predictors()) %>%  
+\ \ step_dummy(all_nominal_predictors()) %>%  
+\ \ step_zv(all_predictors()) %>%  
+\ \ step_log(distance) %>% 
+\ \ step_normalize(all_numeric_predictors())
+:::
+
+Save the recipe we like
+
+<br>
+
+## Using a recipe with workflows
+
+recipes are can be used with {workflows} to "combine" it with a model
+
+<br>
+
+:::{style="font-family: 'Source Code Pro', monospace; font-size: 0.8em;"}
+wf_spec <- workflow() %>%  
+\ \ [add_recipe(rec_spec)]{style="color: #CA225E;"} %>%  
+\ \ add_model(linear_reg())
+:::
+
+## Recipes are estimated
+
+<br>
+
+Every preprocessing step in a recipe that involved calculations uses the *training* set. For example:
+
+- Levels of a factor
+- Determination of zero-variance
+- Normalization
+- Feature extraction
+
+Once a a recipe is added to a workflow, this occurs when `fit()` is called.
+
+
+## Debugging a recipe
+
+- Typically, you will want to use a workflow to estimate and apply a recipe.
+
+. . .
+
+- If you have an error and need to debug your recipe, the original recipe object (e.g. `rec_spec`) can be estimated manually with a function called `prep()`. It is analogous to `fit()`. See [TMwR section 16.4](https://www.tmwr.org/dimensionality.html#recipe-functions)
+
+. . .
+
+- Another function (`bake()`) is analogous to `predict()`, and gives you the processed data back.
+
+. . .
+
+- The `tidy()` function can be used to get specific results from the recipe.
```

Comment:
```suggestion

```

I feel this may be too much for one slide, especially before folks get a sense of what `bake()` does and can differentiate that from this description. Maybe introduce after the prepping + baking exercises?

This may be a lateral move--if you're hesitant to make this change, totally fine with me to wait and see if Hannah thinks this is a good idea.

Reply:
done in [`54f4416` (#233)](https://github.com/tidymodels/workshops/pull/233/commits/54f4416aef25bd1503ac6c08697b8fb36cf2023c)